### DISCUSSION TITLE
Markdown Support for Trilium with Obsidian Compatibility

### DISCUSSION THREAD

**Discussion by Elia1996** [TriliumNext/Trilium#4027] [Ideas]
# Markdown Support for Trilium with Obsidian Compatibility

**Dear Trilium Developers,**

I am writing to propose **adding Markdown file support** to Trilium, our note-taking application. This feature is crucial for our company as we are currently evaluating Trilium as our preferred solution. The ability to **store notes in Markdown format**, similar to Obsidian, could be useful for our workflow and knowledge management.

## Benefits:
- **Interoperability**: Markdown is widely adopted and provides compatibility with various tools and platforms. By supporting Markdown in Trilium, we can ensure **seamless integration** with other Markdown-based tools and facilitate **efficient collaboration**.

## Implementation:
- **File Storage and Synchronization**: To accommodate Markdown files alongside the existing database structure, we would need to modify Trilium's data model. This will enable **easy synchronization** and provide users with **direct access to their notes** even outside the Trilium application.

- **Rendering and Editing**: Upgrading Trilium's rich text editor to support Markdown rendering and editing will allow users to **work with Markdown seamlessly** and switch between Markdown and rich text views effortlessly.

- **Two-Way Conversion**: To ensure a smooth transition and compatibility, it is essential to provide **two-way conversion** between existing rich text notes and Markdown format. This will enable users to **switch between the two formats** based on their preferences and requirements.

- **User Interface**: Minor adjustments to the user interface, such as adding a Markdown syntax guide and highlighting, will enhance the **user experience** for individuals already familiar with Markdown-based tools like Obsidian.

- **Python Compatibility**: Markdown files are easily parsed with Python, which is a widely used programming language. Adding Markdown support in Trilium opens up opportunities for users to **leverage Python-based automation and analysis** on their notes.

I hope to give you some interesting point, do you see it feasible? what do you think about this?

I also search about similar post but there isn't one specific on the topic.

Thank you for your attention.
Sincerely

**Comment by zadam** [comment]
Hello, Trilium is a WYSIWYG application and there's no plan to change this. I believe there are plenty of Markdown note takers out there.

**Comment by Nriver** [comment]
If you are transitioning from Obsidian to Trilium, you can refer to this resource: [Import from Obsidian](https://github.com/Nriver/trilium-py#import-from-obsidian).

Each markdown software has its own implementation of the markdown syntax, resulting in slight variations and incompatibilities between them.

For instance, Obsidian uses a unique relative path format for linking files, which differs from other software. To ensure correct processing, you need to handle it differently. Details can be found here: [Handling Obsidian's relative path format](https://github.com/Nriver/trilium-py/blob/13ba5d2ca5969c3c8338600d17df850423bd2d13/src/trilium_py/client.py#LL776C23-L776C23).

[VNote](https://github.com/vnotex/vnote), another markdown software, has a specific format for manually adjusting image display size. The image link would look something like this: `![](_v_images/xxx.png =700x)`. This format is not recognized by other software. So, another special handle is required. More information can be found here: [Modifying image display size in VNote](https://github.com/Nriver/trilium-py/blob/13ba5d2ca5969c3c8338600d17df850423bd2d13/src/trilium_py/client.py#LL746C6-L746C6).
